source/views/settings/icon.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import React from 'react'
+import {ScrollView, Image, StyleSheet, Text} from 'react-native'
+import * as Icons from 'react-native-alternate-icons'
+import {Section, Cell} from 'react-native-tableview-simple'
+import {Column} from '../components/layout'
+import includes from 'lodash/includes'
+import * as c from '../colors'
+
+const styles = StyleSheet.create({
+  content: {
+    flex: 1,
+    paddingVertical: 10,
+  },
+  title: {
+    fontSize: 16,
+  },
+  icon: {
+    width: 16,
+    height: 16,
+    borderColor: c.black,
+    borderRadius: 5,
+    borderWidth: StyleSheet.hairlineWidth,
+  },
+})
+
+const icons = [
+  {
+    type: 'default',
+    src: require('../../../images/about/IconTrans.png'),
+    title: 'Old Main',
+  },
+  {
+    type: 'icon_type_windmill',
+    src: require('../../../ios/AllAboutOlaf/windmill.png'),
+    title: 'Wind Turbine (Big Ole)',
+  },
+]
+
+export default class IconSettingsView extends React.PureComponent {
+  static navigationOptions = {
+    title: 'App Icon',
+  }
+
+  state: {
+    iconType: string,
+  } = {
+    iconType: '',
+  }
+
+  componentWillMount() {
+    this.getIcon()
+  }
+
+  setIcon(iconType: string) {
+    if (iconType === 'default') {
+      Icons.reset()
+    } else {
+      Icons.setIconName(iconType)
+    }
+
+    this.getIcon()
```

Comment:
We need to PR upstream to allow reset and setIconName to take callbacks

Reply:
Progress has been made at https://github.com/hawkrives/react-native-alternate-icons/tree/hawken

Reply:
You should update this to use my fork maybe so we can continue to prototype; we can eventually switch back once we PR it and it gets approved / released.

Reply:
Mine is published on npm as `@hawkrives/react-native-alternate-icons`